ApiClient/Source/Query.swift

```diff
@@ -29,10 +29,12 @@ import Alamofire
 protocol Query {
     associatedtype Result
     
-    var resource: String { get }
+    var url: URL? { get }
+    var host: String { get }
     var method: HTTPMethod { get }
     var parameters: Parameters { get }
     var headers: HTTPHeaders { get }
     
-    func handleResult(with data: Data) -> Result
+    init(path: String, queryItems: [URLQueryItem])
```

Comment:
why do you add a constructor to a protocol? The constructor and the parameter it takes are specific to each query and there is no point in forcing it to have any specific constructor.